code/modules/power/prototype_shield_emitter.dm

```diff
@@ -0,0 +1,150 @@
+/obj/machinery/power/proto_sh_emitter
+	name = "Prototype Shield Emitter"
+	desc = "This is a Prototype Shield Emitter that create in front of it a box made of shielding elements to protect the station from heat and pressure"
+	icon = 'icons/obj/power.dmi'
+	icon_state = "proto_sh_emitter"
+	anchored = FALSE
+	density = TRUE
+	max_integrity = 350
+	integrity_failure = 0.2
+	circuit = /obj/item/circuitboard/machine/proto_sh_emitter
+	var/list/signs = list()
+	var/is_on = FALSE
+	var/locked = FALSE
+
+/obj/machinery/power/proto_sh_emitter/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/simple_rotation, ROTATION_ALTCLICK | ROTATION_CLOCKWISE | ROTATION_COUNTERCLOCKWISE | ROTATION_VERBS, null, CALLBACK(src, .proc/can_be_rotated))
+
+/obj/machinery/power/proto_sh_emitter/anchored
+	anchored = TRUE
+
+/obj/machinery/power/proto_sh_emitter/Destroy()
+	if(SSticker.IsRoundInProgress())
+		var/turf/T = get_turf(src)
+		message_admins("Prototype Shield Emitter deleted at [ADMIN_VERBOSEJMP(T)]")
+		log_game("Prototype Shield Emitter deleted at [AREACOORD(T)]")
+	for(var/H in signs)
+		qdel(H)
+	return ..()
+
+/obj/machinery/power/proto_sh_emitter/update_icon_state()
+	if(is_on == TRUE)
+		icon_state = "proto_sh_emitter_on"
+	else
+		icon_state = "proto_sh_emitter"
+
+/obj/machinery/power/proto_sh_emitter/proc/can_be_rotated(mob/user,rotation_type)
+	if(anchored)
+		to_chat(user, "<span class='warning'>It is fastened to the floor!</span>")
+		return FALSE
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/wrench_act(mob/living/user, obj/item/I)
+	if(is_on == TRUE)
+		to_chat(user, "<span class='warning'>You have to turn the [src] off first!</span>")
+		return TRUE
+	if(!anchored)
+		anchored = TRUE
+		to_chat(user, "<span class='warning'>You bolt the [src] to the floor!</span>")
+	else
+		anchored = FALSE
+		to_chat(user, "<span class='warning'>You unbolt the [src] from the floor!</span>")
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/process()
+	var/area/a = get_area(src)
+	var/turf/Turf = get_turf(src)
+	if(a.power_equip == FALSE)
```

Comment:
I'm guessing there's probably a signal that gets called when this happens, so it'd be better to use that.

Reply:
maybe but i dont know what the signal is or how to use it